application/helpers/frontend_helper.php

```diff
@@ -1897,20 +1897,30 @@ function doAssessment($surveyid, $returndataonly=false)
                 }
 
                 // If this is a question (and not a survey field, like ID), save asessment value
-                if (!empty($field['title']))
+                if ($field['qid'] > 0)
                 {
                     /**
                      * Allow Plugin to update assessment value
                      */
                     // Prepare Event Info
                     $event = new PluginEvent('afterSurveyQuestionAssessment');
                     $event->set('surveyId', $surveyid);
-                    $event->set('question', $field);
+                    $event->set('lang', $_SESSION['survey_'.$surveyid]['s_lang']);
+                    $event->set('gid', $field);
+                    $event->set('qid', $field['gid']);
+
+                    if (array_key_exists('sqid', $field))
+                    {
+                        $event->set('sqid', $field['sqid']);
+                    }
+
```

Comment:
OK for subquestion with one choice but not for Array/text, Array/number where [createFieldMap](https://github.com/LimeSurvey/LimeSurvey/blob/master/application/helpers/common_helper.php#L2064) are silly .... 

Why not add [fieldname](https://github.com/LimeSurvey/LimeSurvey/blob/master/application/helpers/common_helper.php#L2087) ? It's the DB name + session name